lib/src/framework.dart

```diff
@@ -203,7 +203,7 @@ Calling them outside of build method leads to an unstable state and is therefore
 }
 
 /// The logic and internal state for a [HookWidget]
-abstract class HookState<R, T extends Hook<R>> {
+abstract class HookState<R, T extends Hook<R>> with Diagnosticable {
```

Comment:
Hook probably should be Diagnosticable too.

Reply:
What's the need to make `Hook` diagnosticable?